### PR TITLE
Application Mobility Documentation

### PR DESCRIPTION
**Pull request by tdawe**
# Description
Application Mobility Documentation

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|  https://github.com/dell/csm/issues/350 |

# Checklist:

- [x] Have you run a grammar and spell checks against your submission?
- [x] Have you tested the changes locally?
- [x] Have you tested whether the hyperlinks are working properly?
- [x] Did you add the examples wherever applicable?
- [ ] Have you added high-resolution images?

### CODE REVIEW

File: content/docs/application-mobility/_index.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+weight: 9
+Description: >
+  Application Mobility for Dell Container Storage Modules (CSM)
+---
+
+>> NOTE: This tech-preview release is not intended for use in production environment.
+
+>> NOTE: Application Mobility requires a time-based license. See [Deployment](./deployment) for instructions.
+
+Container Storage Modules for Application Mobility provides Kubernetes administrators the ability to clone their application workloads and data to other clusters, either on-premise or in the cloud.
```
Comment (rsedlock1958) [inline content/docs/application-mobility/_index.md:13]: Change "provides" to "provide"

File: content/docs/application-mobility/_index.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Application Mobility uses [Velero](https://velero.io) and its integration of [Restic](https://restic.net) to copy both application metadata and data to object storage. When a backup is requested, Application Mobility uses the following options to determine how the application data is backed up:
```
Comment (rsedlock1958) [inline content/docs/application-mobility/_index.md:15]: Change "the following" to "these"

File: content/docs/application-mobility/deployment.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+1. Either create a values.yml file or provide the `--set` options to the `helm install` to override default values from the [Configuration](#configuration) section.
+1. Install the helm chart
+    ```
+    helm install -n application-mobility dell/csm-application-mobility
+    ```
+
+
+### Configuration
+
+The following table lists the configurable parameters of the Application Mobility Helm chart and their default values.
```
Comment (rsedlock1958) [inline content/docs/application-mobility/deployment.md:36]: Change "The following" to "This"

File: content/docs/application-mobility/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+* To debug your application that may not be behaving correctly, please reference Kubernetes [troubleshooting applications guide](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-application/). 
+
+* For tips on debugging your cluster, please see this [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-cluster/).
+
+### Why are there error logs about a license?
+
+Application Mobility requires a license in order to function. See the [Deployment](../deployment) instructions for steps to request a license.
+
+There will be errors in the logs about the license for the following cases:
```
Comment (rsedlock1958) [inline content/docs/application-mobility/troubleshooting.md:45]: Change "the following" to "these"

File: content/docs/application-mobility/use_cases.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+linkTitle: "Use Cases"
+weight: 3
+Description: >
+  Use cases for Dell Container Storage Modules (CSM) for Application Mobility
+---
+
+After Application Mobility is installed, the [dellctl CLI](../../cli/) can be used to register clusters and manage backups and restores of applications.
+
+## Backup and Restore an Application
+The following example details the steps when an application in namespace `demo1` is being backed up and then later restored to either the same cluster or another cluster. In this sample, both Application Mobility and Velero are installed in the `application-mobility` namespace.
```
Comment (rsedlock1958) [inline content/docs/application-mobility/use_cases.md:12]: Change "The following" "This"

File: content/docs/application-mobility/use_cases.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+1. If Velero is not installed in the default `velero` namespace, set the following environment variable to the namespace where it is installed:
```
Comment (rsedlock1958) [inline content/docs/application-mobility/use_cases.md:14]: Cjhange "the following" to "this"

File: content/docs/application-mobility/_index.md (hunk 1 above)
Comment (tdawe) [reply content/docs/application-mobility/_index.md:13]: Fixed

File: content/docs/application-mobility/use_cases.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    dellctl restore create restore1 --from-backup backup1 \
+        --namespace-mappings "demo1:restorens1" --namespace application-mobility
+    ```
+1. Monitor the restore status until it is marked as Completed.
+    ```
+    dellctl restore get --namespace application-mobility
+    ```
+
+## Clone an Application
+The following example details the steps when an application in namespace `demo1` is cloned from a source cluster to a target cluster in a single operation. In this sample, both Application Mobility and Velero are installed in the `application-mobility` namespace.
```
Comment (rsedlock1958) [inline content/docs/application-mobility/use_cases.md:37]: Change "The following" to "This"

File: content/docs/application-mobility/_index.md (hunk 2 above)
Comment (tdawe) [reply content/docs/application-mobility/_index.md:15]: Fixed

File: content/docs/application-mobility/use_cases.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+1. If Velero is not installed in the default `velero` namespace, set the following environment variable to the namespace where it is installed:
```
Comment (rsedlock1958) [inline content/docs/application-mobility/use_cases.md:39]: Delete "following"

**Review by rsedlock1958 — COMMENTED** [review]
I have reviewed this. Several edits are required.

File: content/docs/application-mobility/deployment.md (hunk 3 above)
Comment (tdawe) [reply content/docs/application-mobility/deployment.md:36]: Fixed

File: content/docs/application-mobility/troubleshooting.md (hunk 4 above)
Comment (tdawe) [reply content/docs/application-mobility/troubleshooting.md:45]: Fixed

File: content/docs/application-mobility/use_cases.md (hunk 5 above)
Comment (tdawe) [reply content/docs/application-mobility/use_cases.md:12]: Fixed

File: content/docs/application-mobility/use_cases.md (hunk 6 above)
Comment (tdawe) [reply content/docs/application-mobility/use_cases.md:14]: Fixed

File: content/docs/application-mobility/use_cases.md (hunk 7 above)
Comment (tdawe) [reply content/docs/application-mobility/use_cases.md:37]: Fixed

File: content/docs/application-mobility/use_cases.md (hunk 8 above)
Comment (tdawe) [reply content/docs/application-mobility/use_cases.md:39]: Fixed

File: content/docs/application-mobility/_index.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Application Mobility"
+linkTitle: "Application Mobility"
+weight: 9
+Description: >
+  Application Mobility for Dell Container Storage Modules (CSM)
```
Comment (alexemc) [inline content/docs/application-mobility/_index.md:6]: Shouldn't it be "Dell Container Storage Modules (CSM) for Application Mobility" or just "Application Mobility"?

File: content/docs/application-mobility/deployment.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+| Parameter | Description | Default |
+| - | - | - |
+| `replicaCount` | Number of replicas for the Application Mobility controllers | `1` |
+| `image.pullPolicy` | Image pull policy for the Application Mobility controller images | `IfNotPresent` |
+| `controller.image` | Location of the csm-application-mobility Docker image | `dell/csm-application-mobility-controller:v0.1.0` |
+| `cert-manager.enabled` |  If set to true, cert-manager will be installed during Application Mobility installation | `false` |
+| `veleroNamespace` |  If Velero is already installed, set to the namespace where Velero is insatlled | `velero` |
+| `licenseName` |  Name of the Secret that contains the License for CSM Application Mobility | `license` |
+| `objectstore.secretName` |  If velero is already installed on the cluster, specify the name of the secret in velero namespace that has credentials to access object store | ` ` |
```
Comment (alexemc) [inline content/docs/application-mobility/deployment.md:46]: Is this value optional? A column `Required <Yes|No>` would be helpful.

File: content/docs/application-mobility/troubleshooting.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
+---
+title: "Troubleshooting"
+linkTitle: "Troubleshooting"
+weight: 4
+Description: >
+  Troubleshooting guide
```
Comment (alexemc) [inline content/docs/application-mobility/troubleshooting.md:6]: This is inconsistent with other sections description, like "**Deployment for** Dell Container Storage Modules (CSM) for Application Mobility".

**Review by alexemc — CHANGES_REQUESTED** [review]
Apart from some minor comments that I left, I suggest removing redundant "Dell Container Storage Modules (CSM) for" and "CSM for" throughout the App Mobility documentation.

File: content/docs/application-mobility/deployment.md (hunk 10 above)
Comment (tdawe) [reply content/docs/application-mobility/deployment.md:46]: Added Required column

File: content/docs/application-mobility/_index.md (hunk 9 above)
Comment (tdawe) [reply content/docs/application-mobility/_index.md:6]: Updated this and others to just the title of the page to remain consistent

File: content/docs/application-mobility/troubleshooting.md (hunk 11 above)
Comment (tdawe) [reply content/docs/application-mobility/troubleshooting.md:6]: Updated this and others to just the title of the page to remain consistent

**Review by alexemc — DISMISSED** [review]

File: content/docs/application-mobility/use_cases.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+    ```
+1. Register both the source and target clusters
+    ```
+    dellctl cluster add -n cluster1 -f ~/kubeconfigs/cluster1-kubeconfig
+    dellctl cluster add -n cluster2 -f ~/kubeconfigs/cluster2-kubeconfig
+    ```
+1. If the Storage Class name on the target cluster is different than the Storage Class name on the source cluster where the backup was created, a mapping between source and target Storage Class names must be defined. See [Changing PV/PVC Storage Classes](#changing-pvpvc-storage-classes).
+1. Create a Backup by providing a name, the included namespace where the application is installed, the source cluster, and the target cluster and namespace mapping where the application will be restored.
+    ```
+    dellctl backup create backup1 --include-namespaces demo1 --cluster-id cluster1 \
```
Comment (bpjain2004) [inline content/docs/application-mobility/use_cases.md:52]: JFYI: if the the source cluster is on the same host where dellctl is run, then it is really not required to register the source cluster and also not required to use --cluster-id flag(by default the command is run against local cluster). Update the below "dellctl backup get" command accordingly.

Comment (tdawe) [reply content/docs/application-mobility/use_cases.md]: Removed references to registering source cluster and updated the "dellctl backup get" command to monitor for the restore on the target cluster.

**Review by sharmilarama — APPROVED** [review]

**Review by alexemc — APPROVED** [review]

**Review by bpjain2004 — APPROVED** [review]

**Review by rsedlock1958 — APPROVED** [review]

**Pull request merged**